plugins/notifications-backend/src/service/router.ts

```diff
@@ -20,10 +21,9 @@ import {
   notificationsReadPermission,
   notificationsSetReadPermission,
 } from './permissions';
-import { RouterOptions } from './types';
 
 export async function createRouter(
-  options: RouterOptions,
+  options: LegacyPluginEnvironment,
```

Comment:
@mareklibra I have updated options to `LegacyPluginEnvironment` for now as its not compatible with `RouterOptions` anymore as `PluginDatabaseManager` used in both types are are coming from two different versions of `@backstage/backend-common`. Since we are upgrading this repo to backstage 1.22.2 which updates the `@backstage/backend-common` package to 0.20.1, but `@backstage/backend-plugin-manager` is still using 0.19.8. The incompatibility is because of different knex versions used in `PluginDatabaseManager` coming from two different `@backstage/backend-common` versions. @backstage/backend-common-v0.19.8 runs on knex 2.0 whereas the latest runs on 3.0. LMK if you have any other ideas as to how this issue can be handled.

Reply:
After switching to using the `@backstage/backend-dynamic-feature-service`, maybe this change wouldn't even be necessary anymore.